src/platform/qt/Window.cpp

```diff
@@ -580,10 +580,24 @@ std::function<void()> Window::openTView(A... arg) {
 	};
 }
 
+namespace {
+class RefAdaptor
+{
+public:
+	RefAdaptor(CorePointerSource& source) : source(&source) {}
+
+	inline operator CorePointerSource&() { return *source; }
+	inline operator CorePointerSource*() { return source; }
+	inline operator std::shared_ptr<CoreController>() { return *source; }
+
+	CorePointerSource* source;
+};
+}
+
```

Comment:
This is temporary and the next PR removes it. It's only here so I don't have to make all of the changes at the same time.